src/elements/cartoon/cartoonDataTransformer.ts

```diff
@@ -1,94 +1,123 @@
-import type { Breakpoint } from "@guardian/src-foundations";
 import { undefinedDropdownValue } from "../../plugin/helpers/constants";
 import type { FieldNameToValueMap } from "../../plugin/helpers/fieldView";
 import type { Asset } from "../helpers/defaultTransform";
 import type { MainImageData } from "../helpers/types/Media";
 import type { TransformIn, TransformOut } from "../helpers/types/Transform";
 import type { cartoonFields } from "./CartoonSpec";
 
+type ViewportSize = "small" | "medium" | "large"; // This used to be called "breakpoint"
+
+type Image = {
+  mimeType: "image/jpeg" | "image/png" | "image/svg+xml";
+  file: string;
+  width: number | string;
+  height: number | string;
+};
+
+type Variant = {
+  viewportSize: ViewportSize;
+  images: Image[];
+};
+
+type Fields = {
+  variants: Variant[];
+  role?: string;
+  photographer?: string;
+  caption?: string;
+  source?: string;
+  displayCredit?: string;
+};
+
 export type Element = {
   elementType: string;
-  fields: Record<string, string | undefined>;
+  fields: Fields;
   assets: Asset[];
-  elements?: Element[];
 };
 
 export const transformElementIn: TransformIn<
   Element,
   ReturnType<typeof cartoonFields>
-> = ({ fields, elements }) => {
-  const { role, photographer, caption, source } = fields;
+> = ({ fields }) => {
+  const { role, photographer, caption, source, variants } = fields;
 
-  const getImages = (breakpoint: Breakpoint): MainImageData[] => {
-    if (Array.isArray(elements)) {
-      return elements
-        .filter(
-          (element) =>
-            element.elementType === "image" &&
-            element.fields.breakpoint === breakpoint
-        )
-        .map((element) => {
-          return {
-            mediaId: element.fields.mediaId,
-            mediaApiUri: element.fields.mediaApiUri,
-            assets: element.assets,
-            caption: element.fields.caption,
-          };
-        });
-    } else {
-      return [];
-    }
+  const getImages = (viewportSize: ViewportSize): MainImageData[] => {
```

Comment:
Now that we know more about the shape of this data, does it make sense to convert the image fields defined in `CartoonSpec.tsx` from `MainImageData[]` to a data type that more closely follows our Cartoon object model? This would mean `smallImages`/`largeImages` would become a custom field type with a data type of `Image[]` which would make this transformation quite a bit simply.

In order for this to continue working in the demo, we will also need to update the `setMedia` function here: https://github.com/guardian/prosemirror-elements/blob/ee254d818110533e5bd278af3f9bc91b36a34313/demo/index.ts#L456

Reply:
Thanks @rebecca-thompson. I've made this change (using `Image` instead of `MainImageData`) as discussed